client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -46,21 +47,27 @@
     @GuardedBy("$lock")
     private long offset;
     @GuardedBy("$lock")
+    private final long endOffset;
+    @GuardedBy("$lock")
     private boolean receivedEndOfSegment = false;
     @GuardedBy("$lock")
     private boolean receivedTruncated = false;
     @GuardedBy("$lock")
     private CompletableFuture<SegmentRead> outstandingRequest = null;
 
-    SegmentInputStreamImpl(AsyncSegmentInputStream asyncInput, long offset) {
-        this(asyncInput, offset, DEFAULT_BUFFER_SIZE);
+    SegmentInputStreamImpl(AsyncSegmentInputStream asyncInput, long startOffset) {
+        this(asyncInput, startOffset, Long.MAX_VALUE, DEFAULT_BUFFER_SIZE);
```

Comment:
Is `Long.MAX_VALUE` here supposed to represent unbounded? Should we have a constant instead to make it explicit?

Reply:
done.